app/src/main/java/com/github/scribeWizTeam/scribewiz/Fragments/RecFragment.kt

```diff
@@ -103,59 +117,65 @@ class RecFragment(contentLayoutId: Int) : Fragment(contentLayoutId) {
-        mediaRecorder.setOutputFile(getOutputFilePath())
-        //set the encoder
-        mediaRecorder.setAudioEncoder(MediaRecorder.AudioEncoder.AAC)
-
-        try {
-            //prepare the media recorder
-            mediaRecorder.prepare()
-            //start recording
-            mediaRecorder.start()
-            //set recording in progress
-            isRecording = true
-            //change the button text
-            recordButton.text = "Stop Recording"
-
-            //set the timer
-            timer = object : CountDownTimer(MILLISINFUTURE, COUNT_DOWN_INTERVAL) {
-                override fun onTick(millisUntilFinished: Long) {
-                    recordingTimeText.text = ((MILLISINFUTURE - millisUntilFinished) / COUNT_DOWN_INTERVAL).toString()
-                }
-                override fun onFinish() {
-                //do nothing
+        // Initialize the AudioRecord
+        audioRecorder = AudioRecord(audioSource, sampleRateInHz, channelConfig, audioFormat, bufferSize)
+
+        // Set the output file path
+        outputFilePath = getOutputFilePath()
+        outputFile = File(outputFilePath)
+
+        // Start recording
+        audioRecorder.startRecording()
+        isRecording = true
+        recordButton.text = "Stop Recording"
+
+        // Write the audio data to a file
+        val buffer = ByteArray(bufferSize)
+        val outputStream = FileOutputStream(outputFile)
+
+        Thread(Runnable {
+            while (isRecording) {
+                val bytesRead = audioRecorder.read(buffer, 0, bufferSize)
+                if (bytesRead > 0) {
+                    try {
+                        outputStream.write(buffer, 0, bytesRead)
+                    } catch (e: IOException) {
+                        e.printStackTrace()
+                    }
                 }
             }
-            //start the timer
-            timer.start()
+            outputStream.close()
+        }).start()
+
+        // Set the timer
+        timer = object : CountDownTimer(MILLISINFUTURE, COUNT_DOWN_INTERVAL) {
+            override fun onTick(millisUntilFinished: Long) {
+                recordingTimeText.text = ((MILLISINFUTURE - millisUntilFinished) / COUNT_DOWN_INTERVAL).toString()
+            }
 
-        //if there is any error in recording
-        } catch (e: IOException) {
-            Toast.makeText(requireContext(), "Failed to start the recording", Toast.LENGTH_SHORT).show()
-            e.printStackTrace()
+            override fun onFinish() {
+                // Do nothing
+            }
         }
+        // Start the timer
+        timer.start()
     }
-    //method to stop recording
+
+    // Method to stop recording
     private fun stopRecording() {
-        //stop recording
-        mediaRecorder.stop()
-        //release the media recorder
-        mediaRecorder.release()
-        //set recording not in progress
+        // Stop recording
+        audioRecorder.stop()
```

Comment:
Does this have a way to fail? And in that case could still be locking the file? If not, ignore this comment

Reply:
I don't believe this has a way to fail and lock the file, as far as I know.